apps/sms/test/marionette/conversation_test.js

```diff
@@ -491,4 +503,83 @@ marionette('Conversation Panel Tests', function() {
       });
     });
   });
+
+  suite('Add recipients from thread', function() {
+    var conversationView;
+    setup(function() {
+      var thread = ThreadGenerator.generate({
+        numberOfMessages: 1
+      });
+      messagesApp.launch();
+
+      storage.setMessagesStorage([thread], ThreadGenerator.uniqueMessageId);
+      // Set empty contacts store.
+      storage.setContactsStorage();
+      var inboxView = new InboxView(client);
+      conversationView = inboxView.goToConversation(thread.id);
+      messagesApp.Composer.messageInput.sendKeys('message2');
+    });
+
+    test('Sms being written: Message and recipients should be kept',
+    function() {
+      conversationView.showOptions();
+      messagesApp.Menu.selectAppMenuOption('Add recipients');
+      // this also ensure we are on Composer view
+      messagesApp.getRecipient('+123');
+      client.waitFor(function() {
+        return messagesApp.Composer.messageInput.text() === 'message2';
+      });
+    });
+
+    test('Mms being written: all info must be kept', function() {
+      // add a subject
+      messagesApp.showSubject();
+      messagesApp.Composer.subjectInput.sendKeys('subject1');
+      // click 'add Recipients'
+      conversationView.showOptions();
+      messagesApp.Menu.selectAppMenuOption('Add recipients');
+      // this also ensure we are on Composer view
+      messagesApp.getRecipient('+123');
+      client.waitFor(function() {
+        return messagesApp.Composer.messageInput.text() === 'message2';
+      });
+      client.waitFor(function() {
+        return messagesApp.Composer.subjectInput.text() === 'subject1';
+      });
+
+      // let's add a recipients and send the message.
+      // we will have a thread with several recipients
+      messagesApp.addRecipient('+234');
+      messagesApp.send();
+      client.helper.waitForElement(messagesApp.Conversation.message);
+      // write another MMS
+      messagesApp.Composer.messageInput.sendKeys('message3');
+      messagesApp.showSubject();
+      messagesApp.Composer.subjectInput.sendKeys('subject2');
+      // add a blob
+      messagesApp.Composer.attachButton.tap();
+      messagesApp.Menu.selectSystemMenuOption('Messages Activity Caller');
+
+      console.log('here');
+      activityCallerApp.switchTo();
+      console.log('there');
+      activityCallerApp.pickImage();
+      messagesApp.switchTo();
+      // click 'add Recipients'
+      conversationView.showOptions();
+      messagesApp.Menu.selectAppMenuOption('Add recipients');
+      // assert the content
+      messagesApp.getRecipient('+123');
+      messagesApp.getRecipient('+234');
+      client.waitFor(function() {
+        return messagesApp.Composer.messageInput.text() === 'message3';
+      });
+      client.helper.waitForElement(
+        messagesApp.Selectors.Message.imgAttachment
+      );
+      client.waitFor(function() {
+        return messagesApp.Composer.subjectInput.text() === 'subject2';
+      });
+    });
+  });
```

Comment:
I miss the case of adding a recipient with an empty message :) (I think this is failing with your current code)